src/client/apiClient.ts

```diff
@@ -0,0 +1,89 @@
+import * as http from "@actions/http-client";
+import { StartSimulationRequest } from "./requests/startSimulationRequest";
+import { StartSimulationResponse } from "./responses/startSimulationResponse";
+import { TypedResponse } from "@actions/http-client/lib/interfaces";
+import { HttpClientError, HttpCodes } from "@actions/http-client";
+import { RunInformationResponse } from "./responses/runInformationResponse";
+import { SimulationResponse } from "./responses/simulationResponse";
+import { SeriesResponse } from "./responses/seriesResponse";
+import { RequestsSummaryResponse } from "./responses/requestsSummaryResponse";
+import { OutgoingHttpHeaders } from "http";
+
+export interface ApiClient {
+  startSimulation: (simulationId: string, options?: StartSimulationRequest) => Promise<StartSimulationResponse>;
+  getRunInformation: (runId: string) => Promise<RunInformationResponse>;
+  getSimulations: () => Promise<SimulationResponse[]>;
+  abortRun: (runId: string) => Promise<void>;
+  getUserConcurrentMetric: (runId: string, scenario: string) => Promise<SeriesResponse[]>;
+  getRequestsSummary: (runId: string) => Promise<RequestsSummaryResponse>;
+}
+
+export const apiClient = (): ApiClient => {
+  const client = new http.HttpClient();
+  const conf: HttpConf = {
+    // TODO RND-5 configurable
+    baseUrl: "https://cloud.gatling.io/api/public",
+    apiToken: ""
+  };
+  return {
+    startSimulation: (simulationId, options) => {
+      const payload = options ?? {};
+      return postJson(client, conf, "/simulations/start", payload, { simulation: simulationId });
```

Comment:
**non-blocking:** Only because I do prefer when returning the value instead of using the `return` keyword
```suggestion
  return {
    startSimulation: (simulationId, options) => postJson(client, conf, "/simulations/start", options ?? {}, { simulation: simulationId);
```

(matter of taste)